[inductor] Fix CPP wrapper codegen for ExternKernel args

aakhundov opened the pull request:
Summary: We see IR nodes `repr`-ed directly in the CPP wrapper codegen. Recently, this issue has been fixed for the Python wrapper codegen in D52899373 (https://github.com/pytorch/pytorch/pull/117838). Here we extend the fix to CPP wrapper codegen / AOTInductor.

Test Plan:
New unit tests. In OSS:

```
python test/inductor/test_aot_inductor.py -k test_triton_kernel_multi_output_arg
```

```
python test/inductor/test_aot_inductor.py -k test_triton_kernel_extern_kernel_arg
```

Differential Revision: D52936248




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler

oulgen commented:
I mentioned this in a direct message to Adnan but for posterity, it would be good to do a bit of a code clean up in wrapper.py. There are so many semi identical functions between jit inductor and aot inductor, `val_to_arg_str` being a prime suspect.

aakhundov commented:
> I mentioned this in a direct message to Adnan but for posterity, it would be good to do a bit of a code clean up in wrapper.py. There are so many semi identical functions between jit inductor and aot inductor, `val_to_arg_str` being a prime suspect.

Yeah, the wrapper codegen refactoring is planned. cc @desertfire @chenyang78.

aakhundov committed:
[inductor] Fix CPP wrapper codegen for ExternKernel args (#117931)

Summary:

Recently, this has been fixed for the Python wrapper codegen in D52899373 (https://github.com/pytorch/pytorch/pull/117838). Here we extend the fix to CPP wrapper codegen / AOTInductor.

Test Plan:
New unit tests. In OSS:

```
python test/inductor/test_aot_inductor.py -k test_triton_kernel_multi_output_arg
```

```
python test/inductor/test_aot_inductor.py -k test_triton_kernel_extern_kernel_arg
```

Reviewed By: oulgen, chenyang78

Differential Revision: D52936248

chenyang78 commented:
> I mentioned this in a direct message to Adnan but for posterity, it would be good to do a bit of a code clean up in wrapper.py. There are so many semi identical functions between jit inductor and aot inductor, `val_to_arg_str` being a prime suspect.

Yeah, wrapper.py is getting quite messy now. I will start refactoring it very soon, likely this coming week. Thanks.
